app/controllers/admin/merchants_controller.rb

```diff
@@ -6,10 +6,18 @@ def show
   end
 
   def update
-    user_to_be_updated = User.find(params[:id])
-    user_to_be_updated.disable_merchant_items
-    user_to_be_updated.update(role: 0)
-    flash[:notice] = "Merchant #{user_to_be_updated.name} downgraded"
-    redirect_to admin_user_path(user_to_be_updated)
+    if params[:disable]
+      merchant = User.find(params[:id])
+      merchant.disable_merchant_items
+      merchant.update(enabled: false)
+      flash[:merchant_disabled] = "#{merchant.name} is now disabled"
+      redirect_to merchants_path
+    else
```

Comment:
Can we make this an elsif so it's clear that this path is for downgrading a merchant to a user

Reply:
Done!